src/profile_manager/views.py

```diff
@@ -59,19 +77,36 @@ class ProfileListCurrent(ProfileList):
     Arguments:
         ProfileList -- Base class
     """
+    view_type = ViewTypes.CURRENT
 
     # override the staff requirement for ProfileList
     def test_func(self):
         return self.request.user.is_authenticated
 
     def get_queryset(self):
-        profiles_qs = Profile.objects.all_for_active_semester()
+        profiles_qs = Profile.objects.all_for_active_semester().filter(user__is_active=True)
         return self.queryset_append(profiles_qs)
 
-    def get_context_data(self, **kwargs):
-        context = super(ProfileListCurrent, self).get_context_data(**kwargs)
-        context['current_only'] = True
-        return context
+
+@method_decorator(staff_member_required, name='dispatch')
+class ProfileListStaff(ProfileList):
+    view_type = ViewTypes.STAFF
```

Comment:
Nicely done (Inheritance and the enum)

Doesn't Profile List already implement a staff member required test_func?  Why the duplication?